spec/features/merging_spec.rb

```diff
@@ -1,68 +1,254 @@
+        page_has_selector "input[name='request'][value='#{merge_request.id}']", count: 2
+        page_has_selector "input.btn[value='Approve']", count: 1
+        page_has_selector "input[name='decision'][value='approved']", count: 1
+        page_has_selector "input.btn[value='Deny']", count: 1
+        page_has_selector "input[name='decision'][value='denied']", count: 1
+      end
+    end
 
+    def should_commit_merge
       successfully_visits_page entity_path(dest)
 
       expect(dest.reload.relationships.count).to eql 1
       expect(dest.lists.count).to eql 1
       expect(source.reload.is_deleted).to be true
       expect(source.merged_id).to eql dest.id
     end
+
+    def should_not_commit_merge
+      successfully_visits_page entity_path(source)
+
+      expect(dest.reload.relationships.count).to eql 0
+      expect(dest.lists.count).to eql 0
+      expect(source.reload.is_deleted).to be false
+      expect(source.merged_id).to be_nil
+    end
+
+    before do
+      allow(Entity::Search).to receive(:similar_entities).and_return([dest])
+      visit tools_merge_path(mode:    mode,
+                             source:  source&.id,
+                             dest:    dest&.id,
+                             query:   query,
+                             request: merge_request&.id)
+    end
+
+    context 'as a non-admin user' do
+      let(:user) { create(:really_basic_user) }
+
+      context 'searching for merge targets' do
+        let(:mode) { ToolsController::MergeModes::SEARCH }
+        let(:dest_param) {}
+
+        it "allows access" do
+          expect(page).to have_http_status 200
+        end
+      end
+
+      context 'requesting a merge' do
+        let(:mode) { ToolsController::MergeModes::REQUEST }
+        let(:query_param) {}
+
+        it "allows access" do
+          expect(page).to have_http_status 200
+        end
+      end
+
+      context 'executing a merge' do
+        let(:mode) { ToolsController::MergeModes::EXECUTE }
+        let(:query_param) {}
+
+        denies_access
+      end
+
+      context 'reviewing a merge' do
+        let(:mode) { ToolsController::MergeModes::REVIEW }
+        let(:query_param) {}
+
+        denies_access
+      end
+    end
+
+    context 'as an admin' do
+      let(:user) { create(:admin_user) }
+
+      context 'searching for merge targets' do
+        let(:mode) { ToolsController::MergeModes::SEARCH }
+        let(:query) { source.name }
+
+        it "has a search bar to search for more matches" do
+          page_has_selector 'form[action="/tools/merge"]'
+          page_has_selector 'input[@value="search"]'
+          page_has_selector "input[@value='#{source.id}']"
+        end
+
+        it 'shows a table of last search matches' do
+          page_has_selector 'h1', text: "Merge #{source.name} with another person"
+        end
+
+        it "is impossible to test the contents of the table b/c it is in javascript"
```

Comment:
not impossible...just requires getting browser tests to work. one day.

Reply:
`s/impossible/currently impossible` ;)